plugins/http/guest-js/index.ts

```diff
@@ -26,6 +26,38 @@
 
 import { invoke } from "@tauri-apps/api/primitives";
 
+/**
+ * Configuration of a proxy that a Client should pass requests to.
+ *
+ * @since 2.0.0
+ */
+export interface ProxyConfig {
+  /**
+   * Proxy all traffic to the passed URL.
+   */
+  all?: string;
+  /**
+   * Proxy all HTTP traffic to the passed URL.
+   */
+  http?: string;
+  /**
+   * Proxy all HTTPS traffic to the passed URL.
+   */
+  https?: string;
+  /**
+   * Set the `Proxy-Authorization` header using Basic auth.
+   */
+  basicAuth?: {
+    username: string;
+    password: string;
+  };
+  /**
+   * A configuration for filtering out requests that shouldn’t be proxied.
+   * Entries are expected to be comma-separated (whitespace between entries is ignored)
+   */
+  noProxy?: string;
+}
```

Comment:
This interface doesn't enforce using one of `all`, `https` or `http` fields so `basicAuth` or `noProxy` won't have any effect, and also allows using multiple of them at once where on the backend, only one could be used.

I think we can refactor this type to be something like this to enforce these restrictions:
```ts
type Proxy = {
  all: string | ProxyConfig;
} | {
  https: string | ProxyConfig;
} | {
  http: string | ProxyConfig;
} | {};

interface ProxyConfig {
  url: string;
  basicAuth?: {
    username: string;
    password: string;
  };
  noProxy?: string;
}
```

Reply:
`http` and `https` are not mutually exclusive, they can co-exist, I think we can write it this way: `export type Proxy =
  | {
      all: string | ProxyConfig;
    }
  | {
      http: string | ProxyConfig;
    }
  | {
      https: string | ProxyConfig;
    }
  | {
      http: string | ProxyConfig;
      https: string | ProxyConfig;
    };`

Reply:
Yeah, I was under the assumption that only one proxy can exist, then I guess we can just simplify this to 
```ts
Proxy = {
  all?: string | ProxyConfig;
  http?: string | ProxyConfig;
  https?: string | ProxyConfig;
}
```